test/e2e/storage/testsuites/base.go

```diff
@@ -421,6 +424,60 @@ func convertTestConfig(in *PerTestConfig) volume.TestConfig {
 	}
 }
 
+// getSizeRangesIntersection takes two instances of storage size ranges and determines the
+// intersection of the intervals (if it exists) and return the minimum of the intersection
+// to be used as the claim size for the test.
+// if value not set, that means there's no minimum or maximum size limitation and we set default size for it.
+func getSizeRangesIntersection(first volume.SizeRange, second volume.SizeRange) string {
+	var firstMin, firstMax, secondMin, secondMax resource.Quantity
+	var err error
+	//if not set, use default size
+	if len(first.Min) == 0 {
+		first.Min = defaultMinSize
+	}
+	if len(second.Min) == 0 {
+		first.Min = defaultMinSize
+	}
+	if len(first.Max) == 0 {
+		first.Min = defaultMaxSize
+	}
+	if len(second.Max) == 0 {
+		first.Min = defaultMaxSize
+	}
+
+	if firstMin, err = resource.ParseQuantity(first.Min); err != nil {
+		framework.Skipf("Failed parsing quantity %s", first.Min)
```

Comment:
I feel this `Skipf` can lead unexpected skip on conformance tests in long term.
Now we are checking conformance tests should not call any `framework.Skip` functions with https://github.com/kubernetes/kubernetes/blob/master/hack/conformance/check_conformance_test_requirements.go
But the scrip just checks inside of conformance tests only, that means the script cannot cover functions which called from conformance tests like this.
So can we replace this `Skipf` with `Failf`?

Reply:
`Failf` would be wrong for tests which are not part of conformance. That's IMHO the more likely situation, because will there ever be a conformance test that requires such a big volume size that a real driver cannot provide it?

Either way, it looks like it would be better to move `Skipf` or `Failf` out of this function. Make it return a value and an error, then let the caller decide what to do in case of an error. Not sure whether that'll be enough for `check_conformance_test_requirements.go`, but it'll be more obvious at least.